### ISSUE TITLE
Coverage gap: role-driven content-category change not enforced against heading descendants

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#4028]
## Problem

Per [ARIA in HTML](https://w3c.github.io/html-aria/) ("Allowed descendants of ARIA roles" table, an informative section):

> Column 2 identifies the Kinds of content categories each `role` has when it is used on an HTML element.

`alert`, `alertdialog`, `application`, `dialog`, `document`, `listbox`, `log`, `marquee`, `math`, `note`, `status`, `tabpanel`, `timer`, and `toolbar` are all listed with "Kind of content: Flow content". Per [HTML LS §4.3.6](https://html.spec.whatwg.org/multipage/sections.html#the-h1,-h2,-h3,-h4,-h5,-and-h6-elements), the content model of `h1`–`h6` is phrasing content only. An element carrying one of these 14 roles is therefore Flow content and must not appear as a descendant of a heading, even if the underlying HTML element (e.g. `span`) is itself phrasing content.

`permitted-contents` does not currently account for `role`-driven content-category changes, so this goes undetected:

```html
<h1>
  <span role="alert" aria-label="Lorem">Ipsum</span>
</h1>
```

Confirmed via the bench snapshot (`all-rules` config): `tests/external/snapshots/markuplint/html-aria/misc/role-alert-inside-h1-novalid.json` reports `"violations": []`.

## Affected bench fixtures (14, all `html-aria/misc/`)

- `role-alert-inside-h1-novalid.html`
- `role-alertdialog-inside-h1-novalid.html`
- `role-application-inside-h1-novalid.html`
- `role-dialog-inside-h1-novalid.html`
- `role-document-inside-h1-novalid.html`
- `role-listbox-inside-h1-novalid.html`
- `role-log-inside-h1-novalid.html`
- `role-marquee-inside-h1-novalid.html`
- `role-math-inside-h1-novalid.html`
- `role-note-inside-h1-novalid.html`
- `role-status-inside-h1-novalid.html`
- `role-tabpanel-inside-h1-novalid.html`
- `role-timer-inside-h1-novalid.html`
- `role-toolbar-inside-h1-novalid.html`

All new to the nu-validator test corpus as of submodule commit `f84563f` (2026-08-30); they surfaced as fresh `nu-only` entries after a routine submodule bump and are not a markuplint regression.

## Notes for implementation

- Candidate location: `packages/@markuplint/rules/src/permitted-contents/` (`content-model.ts` currently has no `role`-aware branch).
- The ARIA in HTML table covers many more roles than these 14 — worth reading the full "Allowed descendants of ARIA roles" table rather than special-casing just these fixtures, since the same content-category-override mechanism likely applies broadly (e.g. `role=button` restricting descendants, per the spec's own example).
- The cited spec section is explicitly informative; the normative basis is the combination of the HTML LS content-model definitions it references and the WAI-ARIA role definitions, so the exact citation to use in the rule/tests should be re-derived per role rather than pointing only at the informative table.

## Benchmark cross-reference

_Sync'd from `tests/external/snapshots/` (submodule=f84563f, nu-validator@sha256:0e2354380e6b, markuplint=5.0.0-rc.7, generated=2026-09-01 02:57:15 UTC)._

Filter: `^html-aria\/misc\/role-.+-inside-h1-novalid` → **15** fixtures. Verdict tally: match-error=1, match-clean=0, ml-only=0, nu-only=14, nu-over=0.

ARIA in HTML's "Allowed descendants of ARIA roles" table (informative) lists 14 roles (alert/alertdialog/application/dialog/document/listbox/log/marquee/math/note/status/tabpanel/timer/toolbar) as Flow content; HTML LS restricts h1-h6 to phrasing content only. `permitted-contents` has no role-driven content-category override, so these 14 fixtures — new to the nu-validator corpus as of submodule commit f84563f (2026-08-30) — stay nu-only until the rule gains role-awareness.

| fixture | verdict | nu id / ml rule |
| --- | --- | --- |
| `html-aria/misc/role-alert-inside-h1-novalid.html` | nu-only | `nv-bfce13d28185` |
| `html-aria/misc/role-alertdialog-inside-h1-novalid.html` | nu-only | `nv-03bfdbeec87e` |
| `html-aria/misc/role-application-inside-h1-novalid.html` | nu-only | `nv-584f26562285` |
| `html-aria/misc/role-dialog-inside-h1-novalid.html` | nu-only | `nv-8559b0405804` |
| `html-aria/misc/role-document-inside-h1-novalid.html` | nu-only | `nv-c5613fe188f7` |
| `html-aria/misc/role-feed-inside-h1-novalid.html` | match-error | — |
| `html-aria/misc/role-listbox-inside-h1-novalid.html` | nu-only | `nv-a66d5097367e` |
| `html-aria/misc/role-log-inside-h1-novalid.html` | nu-only | `nv-ec7ffd0604e5` |
| `html-aria/misc/role-marquee-inside-h1-novalid.html` | nu-only | `nv-67403fa4d36a` |
| `html-aria/misc/role-math-inside-h1-novalid.html` | nu-only | `nv-4d0a6a305516` |
| `html-aria/misc/role-note-inside-h1-novalid.html` | nu-only | `nv-10634351ee10` |
| `html-aria/misc/role-status-inside-h1-novalid.html` | nu-only | `nv-6d1fe82b5ee5` |
| `html-aria/misc/role-tabpanel-inside-h1-novalid.html` | nu-only | `nv-deeba4a415b0` |
| `html-aria/misc/role-timer-inside-h1-novalid.html` | nu-only | `nv-8c0d1cd269ae` |
| `html-aria/misc/role-toolbar-inside-h1-novalid.html` | nu-only | `nv-f26a6be4abf9` |

Reproduce: see [`bench-triage` skill](https://github.com/markuplint/markuplint/blob/dev/.claude/skills/bench-triage/SKILL.md) for the audit workflow.
